diff --git a/src/daemon/caps.c b/src/daemon/caps.c
index fd135c0..56329ab 100644
--- a/src/daemon/caps.c
+++ b/src/daemon/caps.c
@@ -53,6 +53,8 @@ void pa_drop_root(void) {
     uid = getuid();
     gid = getgid();
 
+#if 0
+
 #if defined(HAVE_SETRESUID)
     pa_assert_se(setresuid(uid, uid, uid) >= 0);
     pa_assert_se(setresgid(gid, gid, gid) >= 0);
@@ -66,6 +68,8 @@ void pa_drop_root(void) {
     pa_assert_se(setegid(gid) >= 0);
 #endif
 
+#endif
+
     pa_assert_se(getuid() == uid);
     pa_assert_se(geteuid() == uid);
     pa_assert_se(getgid() == gid);
diff --git a/src/pulsecore/core-util.c b/src/pulsecore/core-util.c
index 19c89a9..c2b6b58 100644
--- a/src/pulsecore/core-util.c
+++ b/src/pulsecore/core-util.c
@@ -3383,12 +3383,13 @@ size_t pa_pipe_buf(int fd) {
 }
 
 void pa_reset_personality(void) {
-
+    pa_log_info("skipping personality reset due to seccomp restrictions.");
+/*
 #if defined(__linux__) && !defined(__ANDROID__)
     if (personality(PER_LINUX) < 0)
         pa_log_warn("Uh, personality() failed: %s", pa_cstrerror(errno));
 #endif
-
+*/
 }
 
 bool pa_run_from_build_tree(void) {
